app/src/main/java/com/duckduckgo/app/feedback/ui/SurveyActivity.kt

```diff
@@ -0,0 +1,163 @@
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.feedback.ui
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.webkit.*
+import androidx.core.content.ContextCompat
+import androidx.lifecycle.Observer
+import com.duckduckgo.app.browser.R
+import com.duckduckgo.app.feedback.model.Survey
+import com.duckduckgo.app.feedback.ui.SurveyViewModel.Command
+import com.duckduckgo.app.feedback.ui.SurveyViewModel.Command.Close
+import com.duckduckgo.app.feedback.ui.SurveyViewModel.Command.LoadSurvey
+import com.duckduckgo.app.global.DuckDuckGoActivity
+import com.duckduckgo.app.global.view.gone
+import com.duckduckgo.app.global.view.show
+import com.duckduckgo.app.statistics.pixels.Pixel
+import com.duckduckgo.app.statistics.pixels.Pixel.PixelName.*
+import kotlinx.android.synthetic.main.activity_user_survey.*
+import javax.inject.Inject
+
+
+class SurveyActivity : DuckDuckGoActivity() {
+
+    private val viewModel: SurveyViewModel by bindViewModel()
+
+    @Inject
+    lateinit var pixel: Pixel
+
+    @SuppressLint("SetJavaScriptEnabled")
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_user_survey)
+        configureListeners()
+
+        webView.settings.javaScriptEnabled = true
+        webView.setBackgroundColor(ContextCompat.getColor(this, R.color.cornflowerBlue))
+        webView.webViewClient = SurveyWebViewClient()
+
+        if (savedInstanceState == null) {
+            configureObservers()
+            consumeIntentExtra()
+        }
+    }
+
+    private fun consumeIntentExtra() {
+        val survey = intent.getSerializableExtra(SURVEY_EXTRA) as Survey
+        viewModel.start(survey)
+    }
+
+    private fun configureListeners() {
+        dismissButton.setOnClickListener {
+            pixel.fire(SURVEY_SURVEY_DISMISSED)
+            viewModel.onSurveyDismissed()
+        }
+    }
+
+    private fun configureObservers() {
+        viewModel.command.observe(this, Observer {
+            it?.let { command -> processCommand(command) }
+        })
+    }
+
+    private fun processCommand(command: Command) {
+        when (command) {
+            is LoadSurvey -> loadSurvey(command.url)
+            is Command.ShowSurvey -> {
+                showSurvey()
+            }
+            is Command.ShowError -> {
+                showError()
+            }
+            is Close -> finish()
+        }
+    }
+
+    private fun loadSurvey(url: String) {
+        progress.show()
+        webView.loadUrl(url)
+    }
+
+    private fun showSurvey() {
+        progress.gone()
+        webView.show()
+    }
+
+    private fun showError() {
+        progress.gone()
+        webView.gone()
+        errorView.show()
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        webView.saveState(outState)
```

Comment:
Are we going to run into the `TransactionTooLargeException` again by trying to save the `WebView` state this way?

Reply:
I don't think so, we're visiting just a few small pages and this WebView is very short-lived.